x/tokenomics/token_logic_module/validator_distribution_test.go

```diff
@@ -211,59 +165,330 @@ func TestDistributeValidatorRewards_ErrorCases(t *testing.T) {
 	}
 }
 
-func TestDistributeValidatorRewards_PrecisionHandling(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	mockStakingKeeper := mocks.NewMockStakingKeeper(ctrl)
-
-	// Create validators with stakes that don't divide evenly
-	validators := []stakingtypes.Validator{
+// TestDistributeValidatorAndDelegatorRewards tests the combined validator and delegator reward distribution.
+// It verifies correct distribution to both validators and their delegators based on delegation amounts,
+// backward compatibility fallbacks, and precision handling for fractional distributions.
+func TestDistributeValidatorAndDelegatorRewards(t *testing.T) {
+	tests := []struct {
+		name                     string
+		validators               []stakingtypes.Validator
+		delegations              map[string][]stakingtypes.Delegation
```

Comment:
Add a comment specifying the key is the operator address too please.

```suggestion
		validatorToDelegationsMap              map[string][]stakingtypes.Delegation
```